test/AngularSpec.js

```diff
@@ -602,6 +602,21 @@ describe('angular', function() {
       expect(copy(new Number(NaN)).valueOf()).toBeNaN();
       /* eslint-enable */
     });
+
+    it('should copy source until reaching a given max depth', function() {
+      var destinationA = {};
+      var destinationB = {};
+
+      destinationA = copy({a1:1, b1:{b2:{b3:1}}, c1:[1,{c2:1}], d1:{d2:1}}, destinationB, 1);
+      expect(destinationA).toEqual({a1:1, b1:'...', c1:'...', d1:'...'});
```

Comment:
This is not necessary, since `destinationA === destinationB` (and testing that is out-of-scope for this test).